### DISCUSSION TITLE
”每次请求最大段落数“等同于上下文功能吗？

### DISCUSSION THREAD

**Discussion by forkmax** [immersive-translate/immersive-translate#1424] [Help]
### 概述 | Summary

<img width="664" alt="image" src="https://github.com/immersive-translate/immersive-translate/assets/37327100/887d7484-ac95-4314-8793-c5d4cd19fd83">

如图显示，”每次请求最大段落数“的默认数值为“1”。

1. 若是设置为“4”，是否意味着GPT之类的大模型会输入学习4段文字，然后output翻译结果。
这样子就获得了有4段上下文的翻译结果。
2. 若是选择默认为“1”，GPT就只能根据每次输入的1段文字，给出单独1段的翻译结果。而无法学习上下文。

如果我的猜测没有错误，在价格上：
1. 设置为“4”，消耗的总token比较多。
2. 设置为“1”，消耗的总token比较少。

个人理解，请指正。

**Comment by cssmagic** [comment]
我也是用户，我说一下我的理解：

这个选项的作用是批量翻译，每一段的翻译结果都是会用上的。所以它不仅提升整体响应速度，还节省 token。

理论上有一定的参考上下文的效果。不过这个效果是副作用，不是主要目的。

**Reply by theowenyoung** [reply]
你关于节省token的理解是正确的，4段 一起发送，会少一些  token 支出。

但是响应速度其实会变慢，因为同时发送4个单段段请求，速度要远远快于一个4段的请求。

另外就是4段的请求，openai经常会有幻觉，导致无法一一对应。

**Reply by cssmagic** [reply]
是的。我想表达的 “提升整体响应速度” 是相对的。比如：

* 当我们对接口请求速率做了限制的时候，批量翻译应该会更快拿到更多的翻译结果。
* 很多翻译片断是非常短的，比如标题、导航、菜单、目录等。此时批量翻译的总长度其实并不长。
* 限制 “每次请求最大文本长度”，应该可以避免批量翻译每次发出的长度过长。

总的来说，这个产品的配置做得相当细致合理 😄

**Reply by cssmagic** [reply]
> 4段的请求，openai经常会有幻觉，导致无法一一对应。

这个是分割符的问题吗？（比如我提的 #1511 这个 issue）

**Reply by forkmax** [reply]
> 你关于节省token的理解是正确的，4段 一起发送，会少一些 token 支出。
> 
> 但是响应速度其实会变慢，因为同时发送4个单段段请求，速度要远远快于一个4段的请求。
> 
> 另外就是4段的请求，openai经常会有幻觉，导致无法一一对应。

关于AI模型出现幻觉，我有所补充：
即使开发者已经内置了prompt (比如: Please only return translations), 但是AI仍然会返回一些提示语，比如”以下是翻译内容“，”好的，我仅会返回翻译，以下是简体中文的翻译“，”以下是不解释原文的翻译内容“......

**Reply by cssmagic** [reply]
好的，明白了，谢谢回复。
